cell_classification/predict.py

```diff
@@ -0,0 +1,375 @@
+import h5py
+import toml
+import argparse
+import pickle
+import os
+from sklearn.metrics import roc_curve, auc, confusion_matrix
+from model_builder import ModelBuilder
+import numpy as np
+from tqdm import tqdm
+from plot_utils import plot_average_roc, plot_metrics_against_threshold
+import toml
+from copy import deepcopy
+from joblib import Parallel, delayed
+import pandas as pd
+import tables
+
+
+def load_model_and_val_data(params):
+    """Load model and validation data from params dict
+    Args:
+        params (dict):
+            dictionary containing model and validation data paths
+    Returns:
+        model (ModelBuilder):
+            trained model
+        val_data (tf.data.Dataset):
+            validation dataset
+    """
+    params["eval"] = True
+    model = ModelBuilder(params)
+    model.prep_data()
+    model.load_model(params["model_path"])
+    val_dset = iter(model.validation_dataset)
+    return model, val_dset
+
+
+def process_to_cells(sample):
```

Comment:
This can be split into two functions
1) A general purpose function which computes the average value of a probability mask within each of the segmented objects provided in a segmentation mask. It returns the results in a pandas df
2) A metrics-specific function which appends the predicted activity next to the ground-truth activity in the activity df